pkg/kubernetes/pod/v1alpha1/kubernetes.go

```diff
@@ -53,28 +63,38 @@ func (k *kubeclient) withDefaults() {
 			return cli.CoreV1().Pods(namespace).List(opts)
 		}
 	}
+	if k.del == nil {
+		k.del = func(cli *clientset.Clientset, namespace, name string, opts *metav1.DeleteOptions) error {
+			return cli.CoreV1().Pods(namespace).Delete(name, opts)
+		}
+	}
+	if k.get == nil {
+		k.get = func(cli *clientset.Clientset, namespace, name string, opts metav1.GetOptions) (*v1.Pod, error) {
+			return cli.CoreV1().Pods(namespace).Get(name, opts)
+		}
+	}
 }
 
 // WithNamespace sets the kubernetes client against
 // the provided namespace
 func WithNamespace(namespace string) kubeclientBuildOption {
-	return func(k *kubeclient) {
+	return func(k *Kubeclient) {
 		k.namespace = namespace
 	}
 }
 
 // WithClientSet sets the kubernetes client against
-// the kubeclient instance
+// the Kubeclient instance
 func WithClientSet(c *clientset.Clientset) kubeclientBuildOption {
-	return func(k *kubeclient) {
+	return func(k *Kubeclient) {
 		k.clientset = c
 	}
 }
 
-// KubeClient returns a new instance of kubeclient meant for
+// KubeClient returns a new instance of Kubeclient meant for
 // cstor volume replica operations
-func KubeClient(opts ...kubeclientBuildOption) *kubeclient {
-	k := &kubeclient{}
+func KubeClient(opts ...kubeclientBuildOption) *Kubeclient {
```

Comment:
can you please rename it as `NewKubeClient()`

Reply:
Please refer to my earlier comments